### ISSUE TITLE
Hash characters are not percent-encoded in URL

### ISSUE THREAD

**Issue by tossj** [dawnlabs/carbon#234]
## Expected Behavior
When copying the URL of a picture the hash characters that may present in some code languages are not percent-encoded. As a result, visiting the URL results in missing code because the browser has interpreted the hash characters as the URL's hash, and not data belonging to the `code` query parameter.

This error has been observed in both Firefox 58.0.2 and Microsoft Edge.

In Python the hash character is used to begin comments, and in some cases at the beginning of a file in a shebang (`#!`). Here is example code and a URL that cannot be revisited because the hash character at the beginning of the code is not percent-encoded within the URL.

```python
#!/usr/bin/env python3

if __name__ == '__main__':
    print('hello world')
```
URL from https://carbon.now.sh:

https://carbon.now.sh/?bg=rgba(171,%20184,%20195,%201)&t=dracula&l=python&ds=false&wc=false&wa=true&pv=64px&ph=32px&ln=true&code=#!/usr/bin/env%20python3%0A%0Aif%20__name__%20==%20'__main__':%0A%20%20%20%20print('hello%20world')
## Actual Behavior
Hash characters (`#`) appearing in code should be percent-encoded to `%23` within the URL.

Alternatively, the entire code can be base64-encoded and placed into the URL. This would remove this hash problem and also reduce the size of the URL.

**Comment by mfix22** [comment]
@briandennis is this a duplicate of your recent fix that is yet to be deployed?

**Comment by molnarmark** [comment]
[This one.](https://github.com/dawnlabs/carbon/issues/229)

**Comment by briandennis** [comment]
Yep, as @molnarmark mentioned this was addressed with #229 and #230. It will be pushed live soon 👍
